packages/aws-cdk/lib/api/hotswap/lambda-functions.ts

```diff
@@ -24,7 +23,8 @@ export async function isHotswappableLambdaFunctionChange(
       return ChangeHotswapImpact.REQUIRES_FULL_DEPLOYMENT;
     }
 
-    const functionName = await establishFunctionPhysicalName(logicalId, change, evaluateCfnTemplate);
+    const functionNameInCfnTemplate = change.newValue?.Properties?.FunctionName;
```

Comment:
1. Let's inline `functionNameInCfnTemplate`, doesn't seem like it's bringing much value.
2. I thought `change.newValue` is never `undefined`, is that not correct...?